src/server.rs

```diff
@@ -103,52 +104,42 @@ fn init_async_part(paths: EwwPaths, ui_send: UnboundedSender<app::DaemonCommand>
     });
 }
 
-#[cfg(not(target_os = "linux"))]
-async fn run_filewatch<P: AsRef<Path>>(
-    config_file_path: P,
-    scss_file_path: P,
-    evt_send: UnboundedSender<app::DaemonCommand>,
-) -> Result<()> {
-    Ok(())
-}
-
 /// Watch configuration files for changes, sending reload events to the eww app when the files change.
-#[cfg(target_os = "linux")]
-async fn run_filewatch<P: AsRef<Path>>(
-    config_file_path: P,
-    scss_file_path: P,
-    evt_send: UnboundedSender<app::DaemonCommand>,
-) -> Result<()> {
-    let mut inotify = inotify::Inotify::init().context("Failed to initialize inotify")?;
-    let config_file_descriptor = inotify
-        .add_watch(config_file_path.as_ref(), inotify::WatchMask::MODIFY)
-        .context("Failed to add inotify watch for config file")?;
-    let scss_file_descriptor = inotify
-        .add_watch(scss_file_path.as_ref(), inotify::WatchMask::MODIFY)
-        .context("Failed to add inotify watch for scss file")?;
-
-    let mut buffer = [0; 1024];
-    let mut event_stream = inotify.event_stream(&mut buffer)?;
+async fn run_filewatch<P: AsRef<Path>>(config_dir: P, evt_send: UnboundedSender<app::DaemonCommand>) -> Result<()> {
+
+    use notify::Watcher;
+
+    let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+    let mut watcher: notify::RecommendedWatcher = notify::Watcher::new_immediate(move |res: notify::Result<notify::Event>| match res {
+        Ok(event) => {
+            tx.send(event.paths).unwrap();
```

Comment:
Not a fan of this unwrap, as that may _actually_ fail if the loop_select_exiting exits

Reply:
I can't use a ? here, I think, so what do I do?

Reply:
```
if let Err(err) = tx.send(event.paths) {
  log::warn!("Error forwarding file update event: {:?}", err);
}
```